contracts/token/ERC721/ERC721.sol

```diff
@@ -284,6 +282,28 @@ contract ERC721 is ERC165, IERC721 {
         bytes4 retval = IERC721Receiver(to).onERC721Received(msg.sender, from, tokenId, _data);
         return (retval == _ERC721_RECEIVED);
     }
+    /**
+    * @dev Internal function that safely transfers the ownership of a given token ID to another address
+     * @param from current owner of the token
+     * @param to address to receive the ownership of the given token ID
+     * @param tokenId uint256 ID of the token to be transferred
+     * @param _data bytes data to send along with a safe transfer check
+     */
+    function _safeTransferFrom(address from, address to, uint256 tokenId, bytes memory _data) internal {
+        require(_checkOnERC721Received(from, to, tokenId, _data));
```

Comment:
We should add a revert reason here, as well as in the equivalent `require` in `_safeMint`. You can copy the one that was in `safeTransferFrom`.